tests/cpptests/test_cpp_hybrid_defaults.cpp

```diff
@@ -304,3 +291,59 @@ TEST_F(HybridDefaultsTest, testKAlreadyWithinWindow) {
   ASSERT_EQ(8, vq->knn.k) << "Expected K to remain 8, got " << vq->knn.k;
   ASSERT_EQ(20, parseCtx.hybridParams->scoringCtx->rrfCtx.window);
 }
+
+// Test LINEAR with explicit WINDOW parameter
+TEST_F(HybridDefaultsTest, testLinearExplicitWindow) {
+  RMCK::ArgvList args(ctx, "FT.HYBRID", index_name.c_str(),
+                      "SEARCH", "hello", "VSIM", "@vector", TEST_BLOB_DATA,
+                      "COMBINE", "LINEAR", "6", "ALPHA", "0.6", "BETA", "0.4", "WINDOW", "30");
+
+  parseCommand(args);
+  ASSERT_EQ(parseCtx.hybridParams->scoringCtx->scoringType, HYBRID_SCORING_LINEAR);
+  ASSERT_EQ(30, parseCtx.hybridParams->scoringCtx->linearCtx.window)
+      << "Expected LINEAR window=30, got " << parseCtx.hybridParams->scoringCtx->linearCtx.window;
+
+  // K should remain at default since LINEAR doesn't apply K≤WINDOW constraint
```

Comment:
Is there a difference between RRF and LINEAR? Function `applyLimitKNNFallback` doesn't differntiate?

Reply:
this will be removed because this limit ,K relation is being removed in #6930

Reply:
What about relation between K and WINDOW - not the same for RRF and Linear?

Reply:
The comment says
>  // K should remain at default since LINEAR doesn't apply K≤WINDOW constraint

The relation is
KNN K <= WINDOW
But why should it be different between RRFand LINEAR?

Reply:
I did the change that applies to all